### PR TITLE
Add toolchain subcommand

### PR DESCRIPTION
**Pull request by nhynes**

### CODE REVIEW

File: src/dirs.rs [hunk 1]
```diff
@@ -12,18 +14,39 @@ pub fn is_absolute_path<S: Into<std::ffi::OsString>>(path: S) -> Option<PathBuf>
...
+}
+
+pub fn cache_dir() -> PathBuf {
+    env::var_os("XDG_CACHE_HOME")
+        .and_then(is_absolute_path)
+        .unwrap_or_else(|| home_dir().join(".cache"))
+}
+
+pub fn bin_dir() -> PathBuf {
+    env::var_os("XDG_BIN_HOME")
```
Comment (Yawning) [inline src/dirs.rs]: This is non-standard.  The proposal was never ratified.

File: src/main.rs [hunk 2]
```diff
@@ -87,10 +105,23 @@ fn main() {
         warn!("could not enable telemetry: {}", err);
     };
 
+    // Handle `oasis chain` before args are parsed so that we can get the raw
+    // args to pass to `oasis-chain`. Among other things, Clap won't allow
+    // collecting unknown flags.
```
Comment (Yawning) [inline src/main.rs]: It doesn't support `--`?  That's been part of IEEE Std 1003.1 forever.

Comment (nhynes) [reply src/main.rs]: it does, but you can't have unknown args _without_ `--` https://github.com/clap-rs/clap/issues/873#issuecomment-436546860

File: src/dirs.rs [hunk 3]
```diff
@@ -12,18 +14,39 @@ pub fn is_absolute_path<S: Into<std::ffi::OsString>>(path: S) -> Option<PathBuf>
...
+    env::var_os("HOME").is_some()
 }
 
-pub fn config_dir() -> Option<PathBuf> {
+pub fn home_dir() -> PathBuf {
+    PathBuf::from(env::var_os("HOME").unwrap())
+}
+
+pub fn config_dir() -> PathBuf {
     env::var_os("XDG_CONFIG_HOME")
```
Comment (vishwa-raman) [inline src/dirs.rs]: I read only `XDG_RUNTIME_DIR` is expected to be there and the others should be set by the user. We could default to cwd if we can't find any of the `XDG*` directories?

Comment (vishwa-raman) [reply src/dirs.rs]: I guess it is fine as we have meaningful defaults when we can't find any of them. Ignore my comment.

Comment (nhynes) [reply src/dirs.rs]: [spec](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html)

**Pull request merged**